pkg/components/ldap_auth.go

```diff
@@ -0,0 +1,69 @@
+package components
+
+import (
+	"context"
+	"net/http"
+)
+
+type authValidationTransport struct {
+	Wrapped              http.RoundTripper
+	AllowedGroups        []string
+	LdapGroupsHeaderName string
+}
+
+func Contains(s []string, target string) bool {
+	for _, c := range s {
+		if target == c {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *authValidationTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	incomingLdapGroups := req.Header[r.LdapGroupsHeaderName]
+	allowedList := r.AllowedGroups
+	for _, g := range incomingLdapGroups {
+		if Contains(allowedList, g) {
+			continue
+		} else {
+			return newError(http.StatusUnauthorized, "missing required LDAP group"), nil
```

Comment:
suggestion: For the consumer, I think you should add more detail to the error output. Maybe something along the lines of:

```golang
fmt.Sprintf("not matching values for header %s. given values %v. acceptable values %v.", r.HeaderName, r.AcceptableValues, givenValues)
```

Reply:
still need to address this